test/test_browser.py

```diff
@@ -1353,6 +1353,32 @@ def test_fs_idbfs_sync(self, args):
     print('done first half')
     self.btest_exit('fs/test_idbfs_sync.c', cflags=['-lidbfs.js', f'-DSECRET="{secret}"', '-sEXPORTED_FUNCTIONS=_main,_test,_finish', '-lidbfs.js'] + args)
 
+  def test_fs_idbfs_autopersist(self):
+    common_cflags = ['-lidbfs.js', '-sEXPORTED_FUNCTIONS=_main,_test,_finish']
+    test_cases = [
+      'open',
+      'close',
+      'symlink',
+      'unlink',
+      'rename',
+      'mkdir',
+    ]
+
+    def get_defines(current_case_name, phase):
+      return [
+        f'-DTEST_CASE={test_cases.index(current_case_name) if current_case_name in test_cases else -1}',
+        f'-DTEST_PHASE={phase}',
+      ]
+
+    print('preparing')
+    self.btest_exit('fs/test_idbfs_autopersist.c', cflags=common_cflags + get_defines(None, 0) + ['-DTEST_PREPARE'])
```

Comment:
If you use `if/else/endif` as mentioned about you can just remove `get_defines(None, 0)` completely here.

Then you can do `self.cflags += f'-DTEST_CASE={test_cases.index(current_case_name)}` here on this line and remove the `get_defines` function completely.

Then you can just do `cflags=[f'-DTEST_PHASE=1]'` and  `cflags=[f'-DTEST_PHASE=2']` below.